### PR TITLE
Add structure for istioapi in a new repo

### PR DESCRIPTION
**Pull request by mandarjog**
This repo *does not* depend on anything else
Any shared, or user visibile schema, api or vocabulary should be placed
here

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,23 @@
 # istioapi
-API, Config Definitions and Standard Vocabulary definitions for the istio project
+API, Config Schema Definitions and Standard Vocabulary definitions for the istio project
+
+
+Usually definitions are in the form on .proto files.
```
Comment (kyessenov) [inline README.md]: Can you please add that "proto is a schema"?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+
+##
+All other repositories can depend on istioapi
+This repository *will not* depend on any other repos
+
+##
+We may check-in generated .pb.go and .pb.cc files here.
+
+
+## API definitions
```
Comment (kyessenov) [inline README.md]: Shouldn't everything be API? Config schemas are going to be Kube APIs via TRPs.

**Comment by geeknoid** [comment]
Config schemas should not be k8s-specific.

We can use k8s-specific means for distributing config in a k8s cluster, but
the general config model we use should be agnostic of k8s constructs.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
+istio/api
+
+## Schema definitions
+istio/config
+
+## standard vocabulary
```
Comment (douglas-reid) [inline README.md]: Do you want to s/standard/Standard/ ?

**Pull request merged**